lib/services/oidc.go

```diff
@@ -90,6 +93,23 @@ func OIDCClaimsToTraits(claims jose.Claims) map[string][]string {
 	return traits
 }
 
+// GetRedirectURL gets a redirect URL for the given connector. If the connector
+// has a redirect URL which matches the host of the given Proxy address, then
+// that one will be returned. Otherwise, the first URL in the list will be returned.
+func GetRedirectURL(conn types.OIDCConnector, proxyAddr string) string {
```

Comment:
I think it is critical to test this particular function. 

Also, marshalling tests in `teleport.e` would be a good idea too, as well as any necessary updates to `tctl sso oidc configure` and `tctl sso test`.